### PR TITLE
Add XLC to PATH on AIX

### PR DESCRIPTION
**Pull request by curiositycasualty**
ACRONYM

/cc @chef-cookbooks/engineering-services

### CODE REVIEW

File: recipes/_environment.rb [hunk 1]
```diff
@@ -110,6 +110,10 @@
   end
 else
 
+  if aix?
+    omnibus_env['PATH'].unshift('/opt/IBM/xlC/13.1.0/bin:/opt/IBM/xlC/13.1.0/bin')
+  end
+
```
Comment (schisamo) [inline recipes/_environment.rb]: I think we want this set where the solaris paths are:

```ruby
omnibus_env['PATH'] << "/opt/#{node['omnibus']['toolchain_name']}/embedded/bin" if omnibus_toolchain_enabled?
omnibus_env['PATH'] << '/usr/local/bin'

if aix?
  omnibus_env['PATH'] << '/opt/IBM/xlC/13.1.0/bin'
elsif solaris_10?
  omnibus_env['PATH'] << '/usr/sfw/bin'
  omnibus_env['PATH'] << '/usr/ccs/bin'
end
```

Comment (curiositycasualty) [inline recipes/_environment.rb]: the XLC paths need to appear before `/opt/#{node['omnibus']['toolchain_name']}/embedded/bin` in PATH. Similarly, `/usr/sfw|ccs/bin` need to appear after.

File: recipes/_environment.rb [hunk 2]
```diff
@@ -151,7 +155,11 @@
       echo "Ruby.........$(ruby --version | head -1)"
       echo "RubyGems.....$(gem --version | head -1)"
       echo "Bundler......$(bundle --version | head -1)"
-      echo "GCC..........$(gcc --version | head -1)"
+      #{ if aix?
+          'echo "GCC..........$(xlc -qversion | head -1)"'
```
Comment (schisamo) [inline recipes/_environment.rb]: Should this say `echo "XLC`? It's not GCC right?

File: recipes/_environment.rb (hunk 1 above)
Comment (schisamo) [inline recipes/_environment.rb]: Why does `/opt/IBM/xlC/13.1.0/bin` have to come first in the `PATH`? Even if that is so you could do:

```ruby
omnibus_env['PATH'] << "/opt/#{node['omnibus']['toolchain_name']}/embedded/bin" if omnibus_toolchain_enabled?
omnibus_env['PATH'] << '/usr/local/bin'

if aix?
  omnibus_env['PATH'].unshift('/opt/IBM/xlC/13.1.0/bin')
elsif solaris_10?
  omnibus_env['PATH'] << '/usr/sfw/bin'
  omnibus_env['PATH'] << '/usr/ccs/bin'
end
```

Comment (curiositycasualty) [inline recipes/_environment.rb]: Made it OS conditional as suggested.

**Pull request merged**